### PR TITLE
fix gcc 11 compilation errors

### PR DESCRIPTION
**Pull request by rongou**
Fix gcc 11 compilation errors.

### CODE REVIEW

File: cpp/src/io/orc/aggregate_orc_metadata.cpp [hunk 1]
```diff
@@ -171,7 +171,7 @@ std::vector<metadata::stripe_source_mapping> aggregate_orc_metadata::select_stri
 
       // Coalesce stripe info at the source file later since that makes downstream processing much
       // easier in impl::read
-      for (const size_t& stripe_idx : user_specified_stripes[src_file_idx]) {
+      for (const size_t stripe_idx : user_specified_stripes[src_file_idx]) {
```
Comment (vuule) [inline cpp/src/io/orc/aggregate_orc_metadata.cpp:174]: https://github.com/rapidsai/cudf/pull/10045 already addresses this one.

**Review by vuule — COMMENTED** [review]
C++ looks good, minus the redundant change with an existing PR.

Please add a description to the PR

File: java/src/main/native/src/CompiledExpression.cpp [hunk 2]
```diff
@@ -59,7 +59,7 @@ class jni_serialized_ast {
     check_for_eof(sizeof(T));
```
Comment (jlowe) [inline java/src/main/native/src/CompiledExpression.cpp:59]: 2022 copyrights

File: cpp/include/cudf/strings/repeat_strings.hpp [hunk 3]
```diff
@@ -18,6 +18,8 @@
 #include <cudf/scalar/scalar.hpp>
 #include <cudf/strings/strings_column_view.hpp>
 
+#include <optional>
```
Comment (davidwendt) [inline cpp/include/cudf/strings/repeat_strings.hpp:21]: The `<optional>` include is being addressed in #10044

File: java/src/main/native/src/CompiledExpression.cpp (hunk 2 above)
Comment (rongou) [reply java/src/main/native/src/CompiledExpression.cpp:59]: Done.

File: cpp/benchmarks/common/generate_benchmark_input.cpp [hunk 4]
```diff
@@ -31,6 +31,7 @@
 
 #include <future>
 #include <memory>
+#include <optional>
```
Comment (robertmaynard) [inline cpp/benchmarks/common/generate_benchmark_input.cpp:34]: Thanks for finding this benchmark issue, I have updated #10044 to address this as well.

File: cpp/include/cudf/strings/repeat_strings.hpp (hunk 3 above)
Comment (rongou) [reply cpp/include/cudf/strings/repeat_strings.hpp]: Reverted.

File: cpp/benchmarks/common/generate_benchmark_input.cpp (hunk 4 above)
Comment (rongou) [reply cpp/benchmarks/common/generate_benchmark_input.cpp]: Reverted.

**Review by jlowe — APPROVED** [review]

**Comment by rongou** [comment]
rerun tests

**Comment by rongou** [comment]
@gpucibot merge

**Review by ttnghia — APPROVED** [review]

**Comment by vuule** [comment]
rerun tests

**Pull request merged**